__tests__/events.ts

```diff
@@ -0,0 +1,236 @@
+        },
+      ],
+    });
+
+    await deleteEvents();
+    await deleteUsers();
+  });
+});
+
+describe("event creation", () => {
+  test("should be able to create an event", async () => {
+    // Create user in the database
+    const user: User = await createUser(testUserInfo);
+    const testServer = await createTestServerWithUserLoggedIn(user);
+    // Create a test client connected to the test server
+    const client = createTestClient(testServer);
+
+    const mutation = gql`
+      mutation(
+        $title: String!
+        $start: String!
+        $end: String!
+        $venue: String!
+        $image_base64: String!
+        $description: String
+      ) {
+        createEvent(
+          title: $title
+          start: $start
+          end: $end
+          venue: $venue
+          image_base64: $image_base64
+          description: $description
+        ) {
+          title
+          organiser {
+            username
+          }
+          start
+          end
+          venue
+          image_url
+          description
+        }
+      }
+    `;
+    const variables = {
+      title: testEventInfo.title,
+      start: testEventInfo.start.toISOString(),
+      end: testEventInfo.end.toISOString(),
+      venue: testEventInfo.venue,
+      image_base64: "",
+      description: testEventInfo.description,
+    };
+    const response: GraphQLResponse = await client.mutate({
+      mutation,
+      variables,
+    });
+    expect(response.data).toEqual({
+      createEvent: {
+        title: testEventInfo.title,
+        organiser: {
+          username: testUserInfo.username,
+        },
+        start: testEventInfo.start.toISOString(),
+        end: testEventInfo.end.toISOString(),
+        venue: testEventInfo.venue,
+        image_url: "",
+        description: testEventInfo.description,
+      },
+    });
+
+    await deleteEvents();
+    await deleteUsers();
+  });
+});
+
+describe("event deletion", () => {
+  test("should be able to delete an event", async () => {
+    // Create user in the database
+    const user: User = await createUser(testUserInfo);
+    const testServer = await createTestServerWithUserLoggedIn(user);
+    // Create a test client connected to the test server
+    const client = createTestClient(testServer);
+    // Create a test event
+    const event: Event = await createEvent(testEventInfo);
+
+    const mutation = gql`
+      mutation($id: ID!) {
+        deleteEvent(id: $id) {
+          id
+        }
+      }
+    `;
+    const response: GraphQLResponse = await client.mutate({
+      mutation,
+      variables: { id: event.id },
+    });
+    expect(response.data).toEqual({
+      deleteEvent: {
+        id: event.id,
+      },
+    });
+
+    await deleteEvents();
+    await deleteUsers();
+  });
+});
+
+describe("event subscriber addition and removal", () => {
+  test("should be able to add/remove current user to/from event subscribers list", async () => {
```

Comment:
better to test `addSubscriber` and `removeSubscriber` mutations separately, so that one's failure does not affect the other. the behaviour now is that if `addSubscriber` fails, `removeSubscriber` does not get tested